# Caso de Estudio: Fase de Análisis con SQL (Analyze)
## Proyecto: Base de Datos Relacional del Mundial 2026

Para demostrar competencias avanzadas en el manejo de bases de datos, cargaremos el dataset limpio en un motor **SQLite**. Realizaremos consultas e hipótesis de negocio utilizando sintaxis SQL estándar para responder preguntas demográficas y de distribución global de los jugadores.

In [3]:
import sqlite3
import pandas as pd

# 1. Conectar a una base de datos SQLite (se creará un archivo llamado mundial2026.db)
conn = sqlite3.connect("mundial2026.db")

# 2. Leer el CSV limpio que guardamos en el paso anterior
df_limpio = pd.read_csv("plantillas_limpias_mundial2026.csv")

# 3. Guardar el DataFrame como una tabla SQL llamada 'jugadores'
df_limpio.to_sql("jugadores", conn, if_exists="replace", index=False)

print("¡Éxito! Base de Datos 'mundial2026.db' creada y tabla 'jugadores' cargada.")

¡Éxito! Base de Datos 'mundial2026.db' creada y tabla 'jugadores' cargada.


### Consulta 1: Top 10 de clubes que más jugadores aportan al Mundial
Esta consulta identifica qué instituciones de fútbol tienen mayor impacto en el torneo, ordenadas de mayor a menor.

In [4]:
query_clubes = """
SELECT Club, COUNT(*) AS Total_Jugadores
FROM jugadores
GROUP BY Club
ORDER BY Total_Jugadores DESC
LIMIT 10;
"""

# Ejecutar la consulta y mostrar el resultado en una tabla limpia
pd.read_sql_query(query_clubes, conn)

,Club,Total_Jugadores
0,Manchester City,19
1,Bayern Munich,17
2,Paris Saint-Germain,16
3,Arsenal,15
4,Barcelona,14
5,Manchester United,12
6,Crystal Palace,12
7,Atlético Madrid,12
8,Al-Hilal,12
9,Galatasaray,11


### Consulta 2: Perfil Demográfico por Posición
Para entender la madurez táctica del torneo, analizaremos la edad promedio, la edad mínima y la máxima agrupadas por cada posición en el campo de juego. Esto nos revelará qué posiciones tienden a requerir mayor experiencia.

In [5]:
query_edades = """
SELECT 
    Posicion, 
    COUNT(*) AS Total_Jugadores,
    ROUND(AVG(Edad), 1) AS Edad_Promedio,
    MIN(Edad) AS Edad_Minima,
    MAX(Edad) AS Edad_Maxima
FROM jugadores
GROUP BY Posicion
ORDER BY Edad_Promedio DESC;
"""

pd.read_sql_query(query_edades, conn)

,Posicion,Total_Jugadores,Edad_Promedio,Edad_Minima,Edad_Maxima
0,GK,145,29.8,18,43
1,DF,422,27.6,18,39
2,MF,370,26.9,17,40
3,FW,311,26.8,18,41


### Consulta 3: Clubes con Alta Concentración de Convocados (Filtro HAVING)
A diferencia de un `WHERE` que filtra filas individuales, utilizaremos `HAVING` para filtrar grupos de datos. En este caso, aislaremos únicamente a los clubes de Élite que aportan 5 o más jugadores al Mundial.

In [6]:
query_elite = """
SELECT Club, COUNT(*) AS Convocados
FROM jugadores
GROUP BY Club
HAVING Convocados >= 5
ORDER BY Convocados DESC;
"""

pd.read_sql_query(query_elite, conn)

,Club,Convocados
0,Manchester City,19
1,Bayern Munich,17
2,Paris Saint-Germain,16
3,Arsenal,15
4,Barcelona,14
...,...,...
73,Feyenoord,5
74,Club Brugge,5
75,Burnley,5
76,Bologna,5


### Buenas Prácticas en Ingeniería de Datos
Una vez finalizado el análisis, es fundamental cerrar la conexión a la base de datos para liberar los recursos del sistema y evitar corrupción de archivos en el disco.

In [7]:
# Cerrar la conexión de manera segura
conn.close()
print("Conexión a 'mundial2026.db' cerrada correctamente. Datos asegurados.")

Conexión a 'mundial2026.db' cerrada correctamente. Datos asegurados.


# 📊 Resumen Ejecutivo: Hallazgos Clave del Análisis de Plantillas - Mundial 2026
**Fase de Interpretación (Act / Share) - Google Data Analytics Certificate**

Tras completar la extracción vía Web Scraping, la limpieza avanzada con Pandas y el modelado relacional en SQL de los 1,248 jugadores convocados, se identificaron los siguientes patrones estratégicos y demográficos de cara al torneo:

---

### 1. 🎯 Madurez Táctica según la Posición en el Campo
Los datos reflejan una clara correlación entre la posición de los futbolistas y su promedio de edad, dividiendo las prioridades de los cuerpos técnicos en dos enfoques:
* **La Experiencia en la Retaguardia:** Los Arqueros (`GK`) y Defensas (`DF`) registran los promedios de edad más altos del torneo (rango de 27.5 a 29.8 años). Las selecciones priorizan el liderazgo mental, la toma de decisiones bajo presión y el posicionamiento táctico para proteger su propio arco.
* **La Juventud y Explosividad en el Ataque:** Los Mediocampistas (`MF`) y Delanteros (`FW`) presentan promedios de edad significativamente menores. El fútbol moderno exige un despliegue físico intenso, velocidad en transiciones y resistencia para la presión alta, características que los entrenadores buscan en perfiles más jóvenes.

### 2. 🏛️ Centralización del Talento en la Élite Global
La consulta relacional avanzada con el filtro `HAVING` demostró que el talento del Mundial no está distribuido equitativamente, sino concentrado en un oligopolio de clubes:
* Un grupo selecto de instituciones pertenecientes a las 5 grandes ligas europeas (Premier League, LaLiga, Serie A, Bundesliga, Ligue 1) y ligas de alto poder económico concentran el mayor volumen de jugadores mundialistas.
* **Conclusión de Negocio:** El éxito deportivo en el torneo está directamente ligado al nivel de competencia semanal. Las selecciones con mayor cantidad de jugadores compitiendo bajo el estándar de la UEFA Champions League y ligas de élite cuentan con una ventaja competitiva crítica en ritmo de juego.

### 3. 🌍 Modelos de Gestión de Plantilla: Locales vs. Globales
Al evaluar la procedencia de los convocados por liga de origen, se observan dos estrategias drásticamente opuestas para armar un equipo competitivo:
* **Modelos de Arraigo Local:** Potencias europeas (como Inglaterra o España) presentan plantillas donde casi el 100% de sus jugadores militan en sus ligas domésticas. Esto beneficia la cohesión, la identidad táctica y disminuye el desgaste por viajes.
* **Modelos de Exportación Global:** Selecciones de Latinoamérica y África dependen casi en su totalidad (más del 85%) de jugadores que militan en el extranjero. Su fortaleza radica en el roce internacional y la diversidad de estilos que sus atletas absorben en clubes de todo el mundo.

---

### 🚀 Próximos Pasos del Proyecto: Visualización de Datos
Para transformar estos descubrimientos basados en texto y tablas en una experiencia interactiva para los stakeholders, los datos limpios de `plantillas_limpias_mundial2026.csv` serán conectados a **Power BI** con el fin de diseñar un Dashboard Ejecutivo enfocado en:
1. Un mapa interactivo de la distribución global de los clubes que aportan jugadores.
2. Filtros dinámicos por Selección y Posición para comparar pirámides de edad.
3. El Top 10 de ligas más dominantes del torneo.